src/controllers/accomodation.controller.js

```diff
@@ -0,0 +1,47 @@
+import response from '../helpers/response.helper';
+import accomodationServices from '../services/accomodation.service';
+import UserServices from '../services/user.service';
+import AccommodationHelper from '../helpers/accomodation.helper';
+
+/**
+* Class for Travel administrator and supplier to deal with Accommodation
+*/
+class AccomodationController {
+  /**
+   * This function helps to create new accomodation
+   * @param {Object} req request
+   * @param {*} res response
+   * @returns {Object} return user response
+   */
+  static async createAccomodation(req, res) {
+    const {
+      accommodationName, description, locationId, category, owner, images, rooms, services, amenities
+    } = req.body;
+    const user = await UserServices.findUserByEmail(req.user.email);
+    const role = user.role.toLowerCase();
+    const supportedRole = await UserServices.getRole(role);
+    const roleId = supportedRole.dataValues.id;
+    if (!((roleId === 6) || (roleId === 3))) {
+      response.errorMessage(res, 'You are not authorized to perform this action', 401, 'error');
+    }
+    const newAccomodation = {
+      name: accommodationName, description, locationId, category, owner
+    };
+    const createdAccommodation = await accomodationServices.createAccomodation(newAccomodation);
+    const accomodationId = createdAccommodation.dataValues.id;
+    const accommodationImages = await AccommodationHelper.saveImages(images, accomodationId);
+    const accommodationRooms = await AccommodationHelper.createRooms(rooms, accomodationId);
+    const accommodationServices = await AccommodationHelper.createAccommodationServices(services, accomodationId);
```

Comment:
Good work benit but what does this function `createAccommodationServices` do is it creating accommodation services ?,surely am not getting what it is doing specifically.

Reply:
An accommodation might be able to provide other services that they provide. like restaurant, club and bars, etc .... @mujohn26 that's why i did it.